src/sounds.cpp

```diff
@@ -388,16 +393,790 @@ std::pair<std::vector<tripoint>, std::vector<tripoint>> sounds::get_monster_soun
+        for( auto &i : sound_effects_p ) {
+            if( ( i.id == id ) && ( i.variant == "default" ) ) {
+                valid_sound_effects.push_back( i );
+            }
+        }
+    }
+    if( valid_sound_effects.empty() ) {
+        return;
+    }
+    int index = rng( 0, valid_sound_effects.size() - 1 );
+    selected_sound_effect = valid_sound_effects[index];
+    //add_msg ( m_warning, _ ( "rng 1: %i" ), index );
+    Mix_Chunk *effect_to_play = selected_sound_effect.chunk;
+    float pitch_random = rng_float( pitch_min, pitch_max );
+    Mix_Chunk *shifted_effect = do_pitch_shift( effect_to_play, pitch_random );
+    Mix_VolumeChunk( shifted_effect,
+                     selected_sound_effect.volume * OPTIONS["SOUND_EFFECT_VOLUME"] * volume / ( 100 * 100 ) );
+    int channel = get_channel( shifted_effect );
+    Mix_SetPosition( channel, angle, 1 );
+}
+
+void sfx::play_ambient_variant_sound( std::string id, std::string variant, int volume, int channel,
+                                      int duration ) {
+    if( volume == 0 ) {
+        return;
+    }
+    std::vector<sound_effect> valid_sound_effects;
+    sound_effect selected_sound_effect;
+    for( auto &i : sound_effects_p ) {
+        if( ( i.id == id ) && ( i.variant == variant ) ) {
+            valid_sound_effects.push_back( i );
+        }
+    }
+    if( valid_sound_effects.empty() ) {
+        for( auto &i : sound_effects_p ) {
+            if( ( i.id == id ) && ( i.variant == "default" ) ) {
+                valid_sound_effects.push_back( i );
+            }
+        }
+    }
+    if( valid_sound_effects.empty() ) {
+        return;
+    }
+    int index = rng( 0, valid_sound_effects.size() - 1 );
+    selected_sound_effect = valid_sound_effects[index];
+    //add_msg ( m_warning, _ ( "rng: %i" ), index );
+    Mix_Chunk *effect_to_play = selected_sound_effect.chunk;
+    Mix_VolumeChunk( effect_to_play,
+                     selected_sound_effect.volume * OPTIONS["SOUND_EFFECT_VOLUME"] * volume / ( 100 * 100 ) );
+    if( Mix_FadeInChannel( channel, effect_to_play, -1, duration ) == -1 ) {
+        dbg( D_ERROR ) << "Failed to play sound effect: " << Mix_GetError();
+    }
+}
+
+void sfx::fade_audio_group( int tag, int duration ) {
+#ifdef SDL_SOUND
+    Mix_FadeOutGroup( tag, duration );
+#else
+    ( void ) tag;
+    ( void ) duration;
+#endif
+}
+
+void sfx::fade_audio_channel( int tag, int duration ) {
+#ifdef SDL_SOUND
+    Mix_FadeOutChannel( tag, duration );
+#else
+    ( void ) tag;
+    ( void ) duration;
+#endif
+}
+
+void sfx::set_group_channels( int from, int to, int tag ) {
+#ifdef SDL_SOUND
+    Mix_GroupChannels( from, to, tag );
+#else
+    ( void ) from;
+    ( void ) to;
+    ( void ) tag;
+#endif
+}
+
+int sfx::is_channel_playing( int channel ) {
+#ifdef SDL_SOUND
+    return Mix_Playing( channel );
+#else
+    ( void ) channel;
+#endif
+}
+
+void sfx::stop_sound_effect_fade( int channel, int duration ) {
+#ifdef SDL_SOUND
```

Comment:
It might be better to wrap all `sfx` functions into this check and provide empty dummy implements on builds without SDL_SOUND. This check is currently not done consistently at all, `play_ambient_variant_sound` contains calls to various `Mix_`* functions without that check, which will not work on no-sound builds.